examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -0,0 +1,436 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <core/CHIPCore.h>
+#include <messaging/ExchangeMgr.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <protocols/echo/Echo.h>
+#include <shell/shell.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/PASESession.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/raw/TCP.h>
+#include <transport/raw/UDP.h>
+
+#include <ChipShellCollection.h>
+
+using namespace chip;
+using namespace Shell;
+using namespace Logging;
+
+#define ECHO_CLIENT_PORT (CHIP_PORT + 1)
```

Comment:
could use `constexpr uint16_t kEchoClientPort`

why is this not CHIP_PORT ?

Reply:
My concern is that the test makes a lot of assumptions regarding ports and setup so it is only applicable for the TE1 use case. For a generic ping/pong app, it would be nice if all remote parameters would be settable. If we can set 'remote address' why not 'remote port' as well.

Reply:
Agree, we can add later if there is a need to listen to different port other than CHIP_PORT on the server app. I think we also need to add configuration option on the server apps (lighting-app) to configure the listen port.

Reply:
Added the option to specify destination port.